Algebra.java

```diff
@@ -1,67 +1,144 @@
+
 // the Java operations a + b, a - b, a * b, a / b, a % b, and without calling 
 // Math.sqrt. All the functions in this class operate on int values and
 // return int values.
 
 public class Algebra {
 	public static void main(String args[]) {
-	    // Tests some of the operations
-	    System.out.println(plus(2,3));   // 2 + 3
-	    System.out.println(minus(7,2));  // 7 - 2
-   		System.out.println(minus(2,7));  // 2 - 7
- 		System.out.println(times(3,4));  // 3 * 4
-   		System.out.println(plus(2,times(4,2)));  // 2 + 4 * 2
-   		System.out.println(pow(5,3));      // 5^3
-   		System.out.println(pow(3,5));      // 3^5
-   		System.out.println(div(12,3));   // 12 / 3    
-   		System.out.println(div(5,5));    // 5 / 5  
-   		System.out.println(div(25,7));   // 25 / 7
-   		System.out.println(mod(25,7));   // 25 % 7
-   		System.out.println(mod(120,6));  // 120 % 6    
-   		System.out.println(sqrt(36));
+		// Tests some of the operations
+		System.out.println(plus(2, 3)); // 2 + 3
+		System.out.println(minus(7, 2)); // 7 - 2
+		System.out.println(minus(2, 7)); // 2 - 7
+		System.out.println(times(3, 4)); // 3 * 4
+		System.out.println(plus(2, times(4, 2))); // 2 + 4 * 2
+		System.out.println(pow(5, 3)); // 5^3
+		System.out.println(pow(3, 5)); // 3^5
+		System.out.println(pow(-2, 2));
+		System.out.println(div(12, 3)); // 12 / 3
+		System.out.println(div(5, 5)); // 5 / 5
+		System.out.println(div(25, 7)); // 25 / 7
+		System.out.println(mod(25, 7)); // 25 % 7
+		System.out.println(mod(120, 6)); // 120 % 6
+		System.out.println(sqrt(36));
 		System.out.println(sqrt(263169));
-   		System.out.println(sqrt(76123));
-	}  
+		System.out.println(sqrt(76123));
+		System.out.println(pow(3, 5)); // 3^5
+		System.out.println(pow(-2, 5));
+	}
 
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int result = x1;
+		if (x2 > 0) {
+			for (int i = 0; i < x2; i++) { // x1 = x2 +x1
+				result++;
+			}
+		} else {
+			for (int i = 0; i > x2; i--) { // x1 = x1 + (-x2)
+				result--;
+			}
+		}
+
+		return result;
 	}
 
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int result = x1;
+		if (x2 > 0) {
+			for (int i = 0; i < x2; i++) { // x1 = x1 -x2
+				result--;
+			}
+		} else {
+			for (int i = 0; i > x2; i--) { // x1 = x1 -(-x2)
+				result++;
+			}
+		}
+		return result;
 	}
 
 	// Returns x1 * x2
 	public static int times(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int temp1 = x1, temp2 = x2;
+		if (x1 == 0 || x2 == 0) {
+			return 0;
+		}
+		if (x1 < 0) {
+			temp1 = minus(0, x1);
+		}
+		if (x2 < 0) {
+			temp2 = minus(0, x2);
+		}
+		int result = temp1;
+		for (int i = 1; i < temp2; i++) {
+			result = plus(result, temp1);
+		}
+		if (!((x1 > 0 && x2 > 0) || (x1 < 0 && x2 < 0))) {
```

Comment:
use De Morgan's theorem (from discrete math) to simplify this.. it is less than ideal